pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -957,7 +959,40 @@ public int getMaxUserNameLength() throws SQLException {
   }
 
   public int getDefaultTransactionIsolation() throws SQLException {
-    return Connection.TRANSACTION_READ_COMMITTED;
+    if (defaultTransactionIsolation == 0) {
+      String sql;
+      sql = "SELECT setting FROM pg_catalog.pg_settings WHERE name='default_transaction_isolation'";
+
+      try (Statement stmt = connection.createStatement()) {
+        try (ResultSet rs = stmt.executeQuery(sql)) {
+          String level = null;
+          if (rs.next()) {
+            level = rs.getString(1);
+          }
+          if (level == null) {
+            throw new PSQLException(
+                GT.tr(
+                    "Unable to determine a value for DefaultTransactionIsolation due to missing "
+                    + "system catalog data."),
+                PSQLState.UNEXPECTED_ERROR);
+          }
+
+          level = level.toUpperCase(Locale.US);
+          if (level.equals("READ COMMITTED")) {
+            defaultTransactionIsolation = Connection.TRANSACTION_READ_COMMITTED;
+          } else if (level.equals("READ UNCOMMITTED")) {
+            defaultTransactionIsolation = Connection.TRANSACTION_READ_UNCOMMITTED;
+          } else if (level.equals("REPEATABLE READ")) {
+            defaultTransactionIsolation = Connection.TRANSACTION_REPEATABLE_READ;
+          } else if (level.equals("SERIALIZABLE")) {
+            defaultTransactionIsolation = Connection.TRANSACTION_SERIALIZABLE;
+          } else {
+            defaultTransactionIsolation = Connection.TRANSACTION_READ_COMMITTED; // Best guess.
```

Comment:
If it does a buest guess here, it should do one above as well instead of throwing an exception.

Reply:
I don't agree. The error is being thrown on a relatively straightforward query. If that fails, something has gone wrong.